Factory/RequestFactory.php

```diff
@@ -50,6 +52,12 @@ public function createOne(array $tokens, $apiUrl, array $options) {
         $query   = HttpQuery::create($tokens);
         $payload = $method === HttpMethods::GET || $method === HttpMethods::DELETE ? null : Payload::create($tokens);
 
+        $options['CURLOPT_HTTPHEADER'] = array_merge(
```

Comment:
I don't like the whole concept for handling HTTP headers right now. That's something about my architecture and not about your PR as you did right.

Let's build a new type HttpHeader. Input is the options array. Output is the options array containing just one entry with the key ```CURLOPT_HTTPHEADER```. First of all we move https://github.com/CircleOfNice/DoctrineRestDriver/blob/master/Types/CurlOptions.php lines 68,69 to the HttpHeader type to make sure that the HttpHeaders are in the right format (```String -> NotNil```).
Afterwards you can put the stuff you are actually doing here into the HttpHeader type. The ResultFactory will just call ```$options = array_merge($options, HttpHeader::create($options);)```.

Finally we need to improve the tests and add a new test for the HttpHeader type.